controller/linodecluster_controller.go

```diff
@@ -219,6 +220,10 @@ func (r *LinodeClusterReconciler) reconcileDelete(ctx context.Context, logger lo
 	if clusterScope.LinodeCluster.Spec.Network.NodeBalancerID == nil && clusterScope.LinodeCluster.Spec.Network.LoadBalancerType != lbTypeDNS {
 		logger.Info("NodeBalancer ID is missing, nothing to do")
 
+		if len(clusterScope.LinodeMachines.Items) > 0 {
+			return errors.New("Waiting for associated LinodeMachine objects to be deleted")
```

Comment:
should this be an error or just a requeue?

Reply:
this method only returns errors - I am letting [this](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodecluster_controller.go#L144-L150) take care of the re-queue

Reply:
ah, I see, it feels a bit weird that we are returning an error just because we aren't ready to delete yet, but that is probably all we can do without changing this to return a requeue request instead